### DISCUSSION TITLE
getStaticPaths and fallback

### DISCUSSION THREAD

**Discussion by kpirumyan** [vercel/next.js#35012] [Help] [answered]
Could someone explain what will happen if I set fallback=true or ='blocking' for [[...slug]] dynamic page and someone try to open, say, tens of thousands of non-existent pages using some script? This will generate countless pages in the build folder, won't it? And wouldn't that lead to a shortage of storage memory resources on the server side?

**Comment by icyJoseph** [comment, marked as answer]
As far as I understand, not quite, but perhaps?

If you set fallback `true`, then `getStaticProps` kicks in, and if the slugs resolve to something, then yes, something will be rendered. However, if the slugs don't match anything, you can return `{notFound: true}` and that'll just do a 404 page.

When I say, the slugs, match or resolve, I mean, will they always return props? If we were talking about posts, or stuff like that, there'll be a limited amount of pages, given by the number of posts. Take this example that could potentially generate infinite:

```tsx
// pre-renders for 0 and 2, but let the runtime generate more
export const getStaticPaths: GetStaticPaths = async () => {

  const paths = [{params:{slug:0}}, {params:{slug:2}}]

  return { paths, fallback: true };
};

export const getStaticProps: GetStaticProps<{isEven: boolean}> =  (context) => {
    const { params } = context;
    const { slug } = params || {};

    if (typeof slug !== "string") return { notFound: true };

   const isEven = Number(slug) % 2 === 0;

   return isEven ? { props: {isEven} } : {notFound: true};
};
```

In this case, only `pages/4`, `pages/6` etc would yield a page, but odd pages would be a 404. Though, the even pages, are infinite.

The question is then, how are you handling the `slug` as an input. Can it potentially generate infinite pages as the snippet above?

**Reply by kpirumyan** [reply]
Thank you so much. {notFound: true} helps to avoid generating infinite pages.

**Reply by larsqa** [reply]
Would this also be true for [`redirect`](https://nextjs.org/docs/api-reference/data-fetching/get-static-props#redirect)? As in, that `redirect` wouldn't render anything.

**Reply by icyJoseph** [reply]
I think you can't return `redirect` from `getStaticProps` at build time, IIRC.

```
Error occurred prerendering page "/posts/1". Read more: https://nextjs.org/docs/messages/prerender-error
Error: `redirect` can not be returned from getStaticProps during prerendering 
```

But, I do think you can at ISR runtime runs of `getStaticProps`.
